server/item/template.go

```diff
@@ -0,0 +1,16 @@
+package item
+
+// Template is an item used in smithing tables to alter tools and armor.
+// They are consumed when used, but can be duplicated using an existing template, its material and diamonds.
+type Template struct {
```

Comment:
I'd call this a SmithingTemplate, since Template alone is rather vague when inside of the item package